File: design/mvc/mvc.py
class Student(object):
    def __init__(self, name):
        self.name = name

    @property
    def data(self):
        return {"name": self.name}


class Display(object):
    def __init__(self, name):
        self.name = name

    def display_json(self, data):
        import json
        data["view"] = self.name
        print(json.dumps(data))


class Control(object):
    def __init__(self, m, v):
        self.m = m
        self.v = v

    def display_student(self):
        self.v.display_json(self.m.data)


def test_mvc():
    student = Student("xiao ming")
    display = Display("json")
    control = Control(student, display)
    control.display_student()

if __name__ == "__main__":
    test_mvc()

File: design/mvc/mvc.go

package main
import (
    "fmt"
    "encoding/json"
)

type Student struct {
    Name string
}

func NewStudent(name string) *Student {
    return & Student {
        Name: name,
    }
}

func (s * Student) Data() map[string]string {
    result := make(map[string]string)
    result["name"] = s.Name
    return result
}


type View struct {
    Name string
}

func NewView(name string) *View {
    return & View{
        Name: name,
    }
}

func (v *View) DisplayJson(data map[string]string) {
    data ["visitor"] = v.Name
    jsonStr, err := json.Marshal(data)
    if err != nil {
        fmt.Println("MapToJsonDemo err: ", err)
        return
    }
    fmt.Println(string(jsonStr))
}

type Control struct {
    M *Student
    V *View
}

func NewControl(s * Student, v *View) *Control {
    return & Control{
        M: s,
        V: v,
    }
}

func (c *Control) DisplayStudentJson() {
    c.V.DisplayJson(c.M.Data())
}

 
func TestMVC() {
    s := NewStudent("xiao ming")
    v := NewView("json")
    c := NewControl(s, v)
    c.DisplayStudentJson()
}
    
func main() {
    TestMVC()
}
